lib/libesp32/ESP32-to-ESP8266-compat/src/esp8266toEsp32.h

```diff
@@ -183,5 +183,19 @@ typedef int SerConfu8;
 
 #define STATION_IF 0
 
+
+// ESP32 needed define to support IDF 5.2 based Arduino builds
+
+// Name has changed
+#ifdef LCD_CAM_LCD_UPDATE_M
+#define LCD_CAM_LCD_UPDATE_REG LCD_CAM_LCD_UPDATE_M
+#endif // LCD_CAM_LCD_UPDATE_M
+
+// IDF 5.2 has changed counting UART channels, SOC_UART_NUM includes now LP UARTS too
+#ifdef SOC_UART_HP_NUM
+#undef SOC_UART_NUM
```

Comment:
Another candidate for deeper investigation. SOC_UART_NUM is used by TasmotaSerial to find number of useable uarts. Tasmota cannot use LP uarts with tasmota code (I think)....

Reply:
Agree, this could fail in edge cases.

Reply:
It behaves like before. With IDF the naming changed. Before IDF 5.2 there was only SOC_UART_NUM (no reference to LP UART). With IDF 5.2 SOC_UART_NUM are ALL UART (including LP UART) So it is redefined to the numbers of "normal" UARTs.

Reply:
The main problem I see here is that SOC_UART_NUM is used in both TasmotaSerial and xsns_53_sml.ino. Changing it in esp8266toesp32.h is no guarantee that it is changed before te previous mentionend files are compiled.

Why not abandon SOC_UART_NUM and use SOC_UART_HP_NUM instead?

Reply:
I'll change it now.

Reply:
Just did a test by using current platform and
```
#define SOC_UART_HP_NUM 2
#ifdef SOC_UART_HP_NUM
#undef SOC_UART_NUM
#define SOC_UART_NUM SOC_UART_HP_NUM
#endif
```
Doesn't work. SOC_UART_NUM at the time TasmotaSerial.cpp is accessing it is still 3 on esp32. 

I'll find a better soultion....

Reply:
SOC_UART_HP_NUM Is the new choise from Espressif. The defines are basically a backpot from 5.2.
One option would be a separate backport header file, that must be included for dependent drivers and could be removed in the future.
But there are many other ways to solve it for sure.